common/workunit/workunit.cpp

```diff
@@ -6930,14 +6936,19 @@ wuTokenStates verifyWorkunitDAToken(const char * distributedAccessToken)
     }
 
     //Verify user matches
-    if (!isEmptyString(cw->queryUser()) || !isEmptyString(tokUser.str()))
+    if (!isEmptyString(cw->queryUser()) && !isEmptyString(tokUser.str()))
     {
-        if (!streq(cw->queryUser(), tokUser.str()))
+        if (!streq(cw->queryUser(), tokUser.str()) || !streq(cw->queryUser(), _user))
         {
-            ERRLOG("verifyWorkunitDAToken : token user does not match workunit");
+            ERRLOG("verifyWorkunitDAToken : token user (%s) does not match WU user (%s) or context user (%s)", tokUser.str(), cw->queryUser(), _user);
```

Comment:
Worth reporting the different errors separately so they can be distinguished from the messages.

Reply:
At some point we will probably want to add both of these errors to the audit log since they suggest someone is trying to circumvent the security.  (but not until logging changes have been merged)

Reply:
They can be distinguished from the message, since all 3 usernames are displayed as part of the error message

Reply:
true, but they cannot be distinguished by grepping the log messages.

If the token does not match the workunit user then that suggests an internal error.
If the token user does not match the requested user that suggests something that should be audited.
So the messages should really have different target audiences.